starter-code/js/index.js

```diff
@@ -1,52 +1,162 @@
-function deleteItem(e){
+var setId = 0;
+
+function deleteRow(e) {
+  console.log(e);
+  var deleteRow = document.getElementById(e);
+  deleteRow.remove();
+  getTotalPrice();
 
 }
 
-function getPriceByProduct(itemNode){
+function updatePriceByProduct(productId) {
 
-}
+  //change totalPrice
+  console.log(productId);
+  var childrenProduct = document.getElementById(productId).children
+  var priceString = childrenProduct.item(1).children.item(0).innerHTML;
+  var price = parseInt(priceString, 10);
+  console.log(price);
 
-function updatePriceByProduct(productPrice, index){
+  var quantityString = childrenProduct.item(2).children.item(1).value;
+  var quantity = parseInt(quantityString, 10);
+  console.log(quantity);
 
+  var totalPrice = childrenProduct.item(3).children.item(0)
+
+  totalPrice.innerHTML = price * quantity + document.getElementById("currency").value;
 }
 
 function getTotalPrice() {
 
-}
+  var products = document.getElementsByClassName("product"); //HTML Collection
+  var productsArr = Array.prototype.slice.call(products);
+  if (productsArr.length == 0) {
+    result = 0;
 
-function createQuantityInput(){
+  } else {
+    var totalPrice = [];
+    productsArr.forEach(element => totalPrice.push(parseInt(element.children.item(3).children.item(0).innerHTML, 10)));
 
-}
+    console.log(totalPrice);
+
+    const reducer = (accumulator, currentValue) => accumulator + currentValue;
+    console.log(totalPrice.reduce(reducer));
 
-function createDeleteButton(){
+    var result = totalPrice.reduce(reducer)
+  }
 
+  document.getElementById("total-price").innerHTML = result + document.getElementById("currency").value;
 }
 
-function createQuantityNode(){
 
-}
 
-function createItemNode(dataType, itemData){
+function createNewItemRow(itemName, itemUnitPrice) {
 
-}
+  var setIdOld = setId;
+  var parentContainer = document.getElementById("container");
+  var productDiv = createNewItem("div", "class", "product");
 
-function createNewItemRow(itemName, itemUnitPrice){
+  productDiv.setAttribute("id", "product-" + setId);
+  parentContainer.appendChild(productDiv);
 
-}
 
-function createNewItem(){
+  var divCounter = 0;
+  while (divCounter < 5) {
+
+    var divInProduct = createNewItem("div", "", "");
+    productDiv.appendChild(divInProduct);
+    divCounter++;
+  }
+
+  // we make HTML collection from all elements with class="product". We need last child from collection.
+
+
+  var LastProductChild = document.getElementsByClassName("product").item(document.getElementsByClassName("product").length - 1);
+
+  var childrenLastproductChild = LastProductChild.children;
+
+  console.log(childrenLastproductChild instanceof HTMLCollection);
+
+  console.log(LastProductChild);
+  console.log(childrenLastproductChild);
+
+
+  var i = 0;
```

Comment:
Creative way to create new product rows :) it's working, but you could have done it with less lines of code. 

Instead of creating all elements needed in the product row with createNewItem() you could have used: 
insertAdjacentHTML("beforeend",`
        <div class="row">
          <ul>
            <li class="name">${newItemName}</li>
            <li>$<span class="price">${newItemPrice}</span></li>
            <li><input class="q" type="number" value="0" name="" id=""></li>
            <li>$<span class="subtotal">0</span></li>
            <li><button class="delete">Delete</button></li>
          </ul>
        </div>
      `)

Than with string interpolation you could insert the information that you want.

Reply:
http://prntscr.com/mlv2z5